hoomd/box.py

```diff
@@ -0,0 +1,393 @@
+        # TODO: check that this works
+        period = make_uint3(value, self.is2D, name='periodicity')
+        self._cpp_obj.setTiltFactors(period)
+
+    @property
+    def lattice_vectors(self):
+        """Box lattice vectors"""
+        return self._lattice_vectors
+
+    @property
+    def volume(self):
+        return self._cpp_obj.getVolume(self.is2D)
+
+    @volume.setter
+    def volume(self, vol):
+        R""" Set the box volume.
+
+        Scale the box to the given volume (or area).
+
+        Args:
+            volume (float): new box volume (area if dimensions=2)
+
+        Returns:
+            A reference to the modified box.
+        """
+        cur_vol = self.volume
+
+        if self.dimension == 3:
+            s = (vol / cur_vol) ** (1.0 / 3.0)
+        else:
+            s = (vol / cur_vol) ** (1.0 / 2.0)
+        self.scale(s)
+
+    @property
+    def matrix(self):
+        Lx, Ly, Lz = self.L
+        xy, xz, yz = self.tilts
+        return np.array([[Lx, xy * Ly, xz * Lz], [0, Ly, yz * Lz], [0, 0, Lz]])
+
+    @matrix.setter
+    def matrix(self, box_matrix):
+        if not isinstance(box_matrix, np.ndarray):
+            box_matrix = np.array(box_matrix)
+        # Error checking
+        if box_matrix.shape == (2, 2):
+            if box_matrix[1, 0] != 0:
+                raise ValueError("Box matrix must be upper triangular.")
+        elif box_matrix.shape == (3, 3):
+            if box_matrix[1, 0] != 0 or any(box_matrix[2, :-1] != 0):
+                raise ValueError("Box matrix must be upper triangular.")
+        else:
+            raise ValueError("Box matrix must be shape (2, 2) or (3, 3)")
+
+        L = np.diag(box_matrix)
+        self.L = L
+        self.xy = box_matrix[0, 1] / L[1]
+        if box_matrix.shape == (3, 3):
+            self.xz = box_matrix[0, 2] / L[2]
+            self.yz = box_matrix[1, 2] / L[2]
+
+    def scale(self, *args):
+        R""" Scale box dimensions.
+
+        Scales the box by the given scale factors.Tilt factors are not
+        modified.
+
+        Args:
+            args (float): scale factors in each dimension. If box is two
+            dimensions then two scale factors is acceptable, but if the box is
+            three dimensions this will error. One scaling factor can be given
+            and used to scale all dimensions.
+
+        Returns:
+            A reference to the modified box.
+        """
+        Nargs = len(args)
+        if Nargs != 1 and Nargs != self.dimension:
+            raise ValueError("The number of scaling factors must be 1 or {}. "
+                             "Given scaling factors {}".format(self.dimension,
+                                                               Nargs))
+        if Nargs == 1:
+            new_L = args[0] * self.L
+        else:
+            new_L = np.array(args) * self.L[:Nargs]
+        self.L = new_L
+        return self
+
+    def wrap(self, v, image=(0, 0, 0)):
```

Comment:
Are wrap, min_image, and make_fraction useful to expose to users?

Reply:
Depends... It would be useful for quick manipulations of particles with the box. Especially once we add user custom `Updaters`, being able to do these easily would be helpful, so someone could for instance right a Python `BoxResizeUpdater` (they shouldn't but it should be possible somewhat easily).

Reply:
In situations like that, methods that work on whole numpy arrays of points would perform better than individual python calls for each operation.

Reply:
Yes it would. That could be something we plan to do. Giving the user more functionality that exists in `freud`'s box. I am not sure of the line to draw here. We could just remove these and suggest users use `freud`, but I feel that it makes sense to give some of that functionality directly to users with our `Box` class.

Reply:
For what it's worth, here's my two cents: I don't think it's worth recreating the features of the freud Box class here. Initialization logic is already being removed from HOOMD, so features that could be used for custom Updater calculations are also IMO fair game to require users to import additional packages. Things like wrapping and image-related math are better fits for an analysis package like freud. I'd prefer to keep the box in HOOMD as stripped down as possible and only promise that it provides a representation of the current box; users can use other packages (like freud) to actually manipulate the system if they need to.

Reply:
The only problem is that for simple casings, should users have to import another package when they have a box object in front of them. Also, at some point we may allow the use of GPU accelerated custom objects in which case `freud` would not be a suitable tool.

On the second point, I could see us deferring those functions until that point at least.